src/rail_map.h

```diff
@@ -530,19 +530,37 @@ static inline void MakeRailNormal(Tile t, Owner o, TrackBits b, RailType r)
 	t.m8() = r;
 }
 
+/**
+ * Sets the exit direction of a rail depot.
+ * @param tile Tile of the depot.
+ * @param dir  Direction of the depot exit.
+ */
+static inline void SetRailDepotExitDirection(Tile tile, DiagDirection dir)
+{
+	assert(IsRailDepotTile(tile));
+	tile.m5() = RAIL_TILE_DEPOT << 6 | dir;
```

Comment:
Technically something like `SB(tile.m5(), 0, 2, dir);` should be used here.
According to the "landscape" specification the `RAIL_TILE_DEPOT` is in the highest two bits, whereas the direction is in the lowest two bits. Practically this means that someone can just start using the four middle bits for something else, and you would be resetting them here. That this is allowed for something like `MakeRailDepot` is due to the fact that you're basically building a completely new object and any new uses of those bits will undoubtedly change the calling signature of `MakeRailDepot`, or they are fine with it being set to 0.

In this case you are now resetting the "path reservation", but it is unclear whether that was intended or an oversight and it happily works. I think it is an oversight, as there is no explicit clearing of the reservation like what happens when removing a single rail tile: https://github.com/OpenTTD/OpenTTD/blob/master/src/rail_cmd.cpp#L656

Reply:
Good point, definitely an oversight. Changed it to SB(...) for both rail and road.